test/e2e/testutils/storage/storage.go

```diff
@@ -54,7 +63,15 @@ func DeployApp(ctx context.Context, config *rest.Config, namespace string) error
 		},
 	}
 
-	if err = cl.Create(ctx, ns); err != nil {
+	if err = retry.OnError(wait.Backoff{
+		Steps:    30,
+		Duration: 100 * time.Millisecond,
+		Factor:   1.1,
+	}, func(error) bool {
+		return apierrors.IsAlreadyExists(err)
+	}, func() error {
+		return cl.Create(ctx, ns)
+	}); err != nil {
```

Comment:
Would personally avoid this and wait for the namespace destruction in the AfterEach block.